### PR TITLE
Classification resnet add dump

### PR DESCRIPTION
**Pull request by tpoisonooo**
增加 resetnet/shufflenet dump 成 .mge 的代码

### CODE REVIEW

File: official/vision/classification/resnet/dump.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    pred_func(data)
+    pred_func.dump(
+        graph_name,
+        arg_names=["data"],
+        optimize_for_inference=True,
+        enable_fuse_conv_bias_nonlinearity=True,
+    )
+
+def main():
+    model = resnet_model.__dict__['resnet18'](pretrained=True)
```
Comment (wjfwzzc) [inline official/vision/classification/resnet/dump.py:35]: 既然是进入公开库，最好是允许传参数，目前写死的`resnet18`不太好……而且感觉这个tool原则上shufflenet什么的都可以用？

File: official/vision/classification/resnet/dump.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    dump_static_graph(model)
```
Comment (wjfwzzc) [inline official/vision/classification/resnet/dump.py:36]: `graph_name`也给个args比较好

File: official/vision/classification/resnet/dump.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import model as resnet_model
+import megengine as mge
+from megengine import jit
+import numpy as np
+
+def dump_static_graph(model, graph_name="model.mge"):
+    model.eval()
+
+    data = mge.Tensor(np.random.random((1, 3, 224, 224)))
```
Comment (wjfwzzc) [inline official/vision/classification/resnet/dump.py:19]: 其实shape也是传个参数比较好吧……用户也不是都用224

**Review by wjfwzzc — CHANGES_REQUESTED** [review]

File: official/vision/classification/resnet/dump.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+# MegEngine is Licensed under the Apache License, Version 2.0 (the "License")
+#
+# Copyright (c) 2014-2021 Megvii Inc. All rights reserved.
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT ARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+# pylint: disable=import-error
```
Comment (wjfwzzc) [inline official/vision/classification/resnet/dump.py:10]: 呃……不要通过这种关掉检查的方式偷家……

Comment (tpoisonooo) [reply official/vision/classification/resnet/dump.py]: 你的代码就是这么写的啊...

Comment (tpoisonooo) [reply official/vision/classification/resnet/dump.py]: 我哪里会，都是从 models 里现学的。

**Comment by tpoisonooo** [comment]
```bash
✗ python3 dump.py --help
usage: dump.py [-h] [-a ARCH] [-s SHAPE [SHAPE ...]] [-o OUTPUT]

MegEngine Classification Dump .mge

optional arguments:
  -h, --help            show this help message and exit
  -a ARCH, --arch ARCH  model architecture (default: resnet18)
  -s SHAPE [SHAPE ...], --shape SHAPE [SHAPE ...]
                        input shape (default: 1 3 224 224)
  -o OUTPUT, --output OUTPUT
                        output filename
```

File: official/vision/classification/dump.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        "-a",
+        "--arch",
+        default="resnet18",
+        help="model architecture (default: resnet18)",
+    )
+    parser.add_argument(
+        "-s",
+        "--shape",
+        type=int,
+        nargs="+",
```
Comment (wjfwzzc) [inline official/vision/classification/dump.py:49]: 看起来这个参数固定消耗4个参数，`nargs=4`

File: official/vision/classification/dump.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT ARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+import argparse
+# pylint: disable=import-error
+import resnet.model as resnet_model
+# pylint: disable=import-error
+import shufflenet.model as snet_model
```
Comment (wjfwzzc) [inline official/vision/classification/dump.py:14]: CI挂在该行末有空格

File: official/vision/classification/dump.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        "-a",
+        "--arch",
+        default="resnet18",
+        help="model architecture (default: resnet18)",
+    )
+    parser.add_argument(
+        "-s",
+        "--shape",
+        type=int,
+        nargs="4",
```
Comment (wjfwzzc) [inline official/vision/classification/dump.py:47]: 你要按照我说的改呀……是`=4`不是`="4"`
改完要跑一遍试试呀……

File: official/vision/classification/dump.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+# MegEngine is Licensed under the Apache License, Version 2.0 (the "License")
+#
+# Copyright (c) 2014-2021 Megvii Inc. All rights reserved.
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT ARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+import argparse
+import resnet.model as resnet_model
```
Comment (wjfwzzc) [inline official/vision/classification/dump.py:11]: 既然说留了，怎么又把`# pylint: disable=import-error`删了……

File: official/vision/classification/dump.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        "--output",
+        type=str,
+        default="model.mge",
+        help="output filename"
+    )
+
+    args = parser.parse_args()
+
+    if 'resnet' in args.arch:
+        model = resnet_model.__dict__[args.arch](pretrained=True)
```
Comment (wjfwzzc) [inline official/vision/classification/dump.py:62]: `model = getattr(resnet_model, args.arch)(pretrained=True)`

File: official/vision/classification/dump.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    elif 'shuffnet' in args.arch:
+        model = snet_model.__name__[args.arch](pretrained=True)
```
Comment (wjfwzzc) [inline official/vision/classification/dump.py:64]: `model = getattr(snet_model, args.arch)(pretrained=True)`

**Review by wjfwzzc — CHANGES_REQUESTED** [review]

**Comment by tpoisonooo** [comment]
嗯，多 PR 两次就会了...

File: official/vision/classification/dump.py [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+import argparse
+# pylint: disable=import-error
+import resnet.model as resnet_model
+# pylint: disable=import-error
+import shufflenet.model as snet_model
+import megengine as mge
+from megengine import jit
+import numpy as np
+import sys
```
Comment (wjfwzzc) [inline official/vision/classification/dump.py]: 请用如下替换，可以过CI
```python
import argparse
import sys

import resnet.model as resnet_model  # pylint: disable=import-error
import shufflenet.model as snet_model  # pylint: disable=import-error

import numpy as np

import megengine as mge
from megengine import jit
```

File: official/vision/classification/dump.py [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
```
Comment (wjfwzzc) [inline official/vision/classification/dump.py:19]: 函数前2个空行

File: official/vision/classification/dump.py [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+        return outputs
+
+    pred_func(data)
+    pred_func.dump(
+        graph_name,
+        arg_names=["data"],
+        optimize_for_inference=True,
+        enable_fuse_conv_bias_nonlinearity=True,
+    )
+
```
Comment (wjfwzzc) [inline official/vision/classification/dump.py:37]: 函数前2个空行

File: official/vision/classification/dump.py [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+    if 'resnet' in args.arch:
+        model = getattr(resnet_model, args.arch)(pretrained=True)
+    elif 'shufflenet' in args.arch:
+        model = getattr(snet_model, args.arch)(pretrained=True)
+    else:
+        print(f'unavailable arch {args.arch}')
+        sys.exit()
+    dump_static_graph(model, args.output, tuple(args.shape))
+
```
Comment (wjfwzzc) [inline official/vision/classification/dump.py:72]: 2个空行

**Review by wjfwzzc — CHANGES_REQUESTED** [review]

**Review by wjfwzzc — APPROVED** [review]
